[APM] Show badge for failed spans in waterfall

sorenlouv opened the pull request:
Closes https://github.com/elastic/kibana/issues/82299
Closes https://github.com/elastic/kibana/issues/109814

**Before**
![image](https://user-images.githubusercontent.com/209966/132457673-80094016-6d1c-43fc-8531-c3d98f25e939.png)


**After**
![image](https://user-images.githubusercontent.com/209966/132456317-0bdfd2c7-88db-41f2-b402-1d9eb1c984a1.png)

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/)
* Commit: c6523d065958abd16b56c50ab376a633dfe0f8db
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/execution/node/418/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/","number":"148096","commit":"c6523d065958abd16b56c50ab376a633dfe0f8db"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148128/)
* Commit: d7c75464ad8ee3dcb056f238ae1b40312d702837
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109812/d7c75464ad8ee3dcb056f238ae1b40312d702837)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f2df9cc9596f4b54bdd728855eb7eaf086463de0) | [after](https://github.com/elastic/kibana/commit/d7c75464ad8ee3dcb056f238ae1b40312d702837) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +892.0B |

### History
* :broken_heart: [Build #148096](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/) failed c6523d065958abd16b56c50ab376a633dfe0f8db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/","number":"148096","commit":"c6523d065958abd16b56c50ab376a633dfe0f8db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148128/","number":"148128","commit":"d7c75464ad8ee3dcb056f238ae1b40312d702837"}]}
PIPELINE-->

formgeist commented on x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/Waterfall/waterfall_item.tsx:
```diff
@@ -216,27 +211,69 @@ export function WaterfallItem({
+    <EuiToolTip
+      content={i18n.translate('xpack.apm.transactionDetails.spanFailure', {
+        defaultMessage:
+          'This span is marked as failed (`event.outcome=failure`)',
```
```suggestion
          'This span is marked as failed (event.outcome=failure)',
```
Don't think the tooltips support code blocks like this.

formgeist commented on x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/Waterfall/waterfall_item.tsx:
```diff
@@ -216,27 +211,69 @@ export function WaterfallItem({
-          </ErrorOverviewLink>
-        ) : null}
+
+        <EventOutcome
+          item={item}
+          transactionErrorCount={transactionErrorCount}
+        />
+
         <Duration item={item} />
         {item.docType === 'span' && <SyncBadge sync={item.doc.span.sync} />}
```
```suggestion
        <Duration item={item} />
        <EventOutcome
          item={item}
          transactionErrorCount={transactionErrorCount}
        />
        {item.docType === 'span' && <SyncBadge sync={item.doc.span.sync} />}
```
I prefer the badges to be displayed after the duration (like other badges we display, which will ensure they line up).

<img width="424" alt="CleanShot 2021-08-25 at 15 17 55@2x" src="https://user-images.githubusercontent.com/4104278/130797629-6ba71244-4a0d-4e02-b6b0-6c6888e41b73.png">

formgeist reviewed (commented):
Just some minor UI suggestions, otherwise LGTM 👍

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/)
* Commit: 5fa51487557485d4802d341cebc3a0519a7be38c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109812/5fa51487557485d4802d341cebc3a0519a7be38c)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2ed3fcdec1128757e5f122c8f8d8e9601d3b584c) | [after](https://github.com/elastic/kibana/commit/5fa51487557485d4802d341cebc3a0519a7be38c) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +272.0B |

### History
* :green_heart: [Build #149253](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/) succeeded 2678d363e9540203fd34f78fe995c9935f6fc4f8
* :broken_heart: [Build #149195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149195/) failed de50f5530718e2e0a40df8a160d8f5e87f66354d
* :green_heart: [Build #148128](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148128/) succeeded d7c75464ad8ee3dcb056f238ae1b40312d702837
* :broken_heart: [Build #148096](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/) failed c6523d065958abd16b56c50ab376a633dfe0f8db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/","number":"148096","commit":"c6523d065958abd16b56c50ab376a633dfe0f8db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148128/","number":"148128","commit":"d7c75464ad8ee3dcb056f238ae1b40312d702837"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149195/","number":"149195","commit":"de50f5530718e2e0a40df8a160d8f5e87f66354d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/","number":"149253","commit":"2678d363e9540203fd34f78fe995c9935f6fc4f8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/","number":"149271","commit":"5fa51487557485d4802d341cebc3a0519a7be38c"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/)
* Commit: 68d6105bc7ed0cea9b3f1e51e412a66385cfac4e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109812/68d6105bc7ed0cea9b3f1e51e412a66385cfac4e)
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/execution/node/428/log)

### History
* :green_heart: [Build #149271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/) succeeded 5fa51487557485d4802d341cebc3a0519a7be38c
* :green_heart: [Build #149253](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/) succeeded 2678d363e9540203fd34f78fe995c9935f6fc4f8
* :broken_heart: [Build #149195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149195/) failed de50f5530718e2e0a40df8a160d8f5e87f66354d
* :green_heart: [Build #148128](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148128/) succeeded d7c75464ad8ee3dcb056f238ae1b40312d702837
* :broken_heart: [Build #148096](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/) failed c6523d065958abd16b56c50ab376a633dfe0f8db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148096/","number":"148096","commit":"c6523d065958abd16b56c50ab376a633dfe0f8db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148128/","number":"148128","commit":"d7c75464ad8ee3dcb056f238ae1b40312d702837"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149195/","number":"149195","commit":"de50f5530718e2e0a40df8a160d8f5e87f66354d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/","number":"149253","commit":"2678d363e9540203fd34f78fe995c9935f6fc4f8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/","number":"149271","commit":"5fa51487557485d4802d341cebc3a0519a7be38c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/","number":"149300","commit":"68d6105bc7ed0cea9b3f1e51e412a66385cfac4e"}]}
PIPELINE-->

dgieselaar commented on x-pack/plugins/apm/public/components/app/service_overview/service_overview_errors_table/index.tsx:
```diff
@@ -213,7 +213,8 @@ export function ServiceOverviewErrorsTable({ serviceName }: Props) {
               noItemsMessage={
-                status === FETCH_STATUS.LOADING
+                status === FETCH_STATUS.LOADING ||
+                status === FETCH_STATUS.NOT_INITIATED
```
Are there scenarios where this will mean the loading spinner will show indefinitely? e.g. by selecting a time range with no data?

sqren replied:
Hmm.. That's a good point. Still, that feels more like an edge case than the existing behaviour

sqren commented on x-pack/plugins/apm/public/components/app/transaction_details/trace_samples_tab.tsx:
```diff
@@ -63,7 +59,6 @@ function TraceSamplesTab({
             urlParams={urlParams}
             waterfall={waterfall}
             isLoading={waterfallStatus === FETCH_STATUS.LOADING}
-            exceedsMax={exceedsMax}
```
This was moved to `waterfall` object to avoid passing it separately through many intermediate components.

sqren commented on x-pack/plugins/apm/public/components/app/transaction_details/use_waterfall_fetcher.ts:
```diff
@@ -13,9 +13,9 @@ import { useTimeRange } from '../../../hooks/use_time_range';
-  errorsPerTransaction: {},
+  errorDocs: [],
+  traceDocs: [],
+  exceedsMax: false,
```
This was unnecessarily nested. Flattening makes it a bit nicer to work with

sqren commented on x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/Waterfall/span_flyout/index.tsx:
```diff
@@ -35,8 +35,9 @@ import { HttpInfoSummaryItem } from '../../../../../../shared/Summary/http_info_
 import { ResponsiveFlyout } from '../ResponsiveFlyout';
 import { SyncBadge } from '../sync_badge';
-import { DatabaseContext } from './database_context';
+import { SpanDatabase } from './span_db';
```
Renamed `DatabaseContext` to `SpanDatabase` since `span.context.db` was renamed to `span.db` long time ago.

sqren commented on x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/Waterfall/waterfall_item.tsx:
```diff
@@ -216,27 +211,46 @@ export function WaterfallItem({
+  if (errorCount > 0) {
+    return (
+      // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+      <div onClick={(e: React.MouseEvent) => e.stopPropagation()}>
```
We want to stop the mouse click from propagating up and opening the flyout.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/)
* Commit: cbc24655478fc6fe624ba1eda0d01c39c01e270a
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/execution/node/428/log)

### History
* :broken_heart: [Build #149300](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/) failed 68d6105bc7ed0cea9b3f1e51e412a66385cfac4e
* :green_heart: [Build #149271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/) succeeded 5fa51487557485d4802d341cebc3a0519a7be38c
* :green_heart: [Build #149253](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/) succeeded 2678d363e9540203fd34f78fe995c9935f6fc4f8
* :broken_heart: [Build #149195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149195/) failed de50f5530718e2e0a40df8a160d8f5e87f66354d
* :green_heart: [Build #148128](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148128/) succeeded d7c75464ad8ee3dcb056f238ae1b40312d702837

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/148128/","number":"148128","commit":"d7c75464ad8ee3dcb056f238ae1b40312d702837"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149195/","number":"149195","commit":"de50f5530718e2e0a40df8a160d8f5e87f66354d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/","number":"149253","commit":"2678d363e9540203fd34f78fe995c9935f6fc4f8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/","number":"149271","commit":"5fa51487557485d4802d341cebc3a0519a7be38c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/","number":"149300","commit":"68d6105bc7ed0cea9b3f1e51e412a66385cfac4e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/","number":"149404","commit":"cbc24655478fc6fe624ba1eda0d01c39c01e270a"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/)
* Commit: cf11c333fe4f1e232207c366e8b4163b8f5c8c76
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109812/cf11c333fe4f1e232207c366e8b4163b8f5c8c76)
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/execution/node/421/log)

### History
* :broken_heart: [Build #149404](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/) failed cbc24655478fc6fe624ba1eda0d01c39c01e270a
* :broken_heart: [Build #149300](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/) failed 68d6105bc7ed0cea9b3f1e51e412a66385cfac4e
* :green_heart: [Build #149271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/) succeeded 5fa51487557485d4802d341cebc3a0519a7be38c
* :green_heart: [Build #149253](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/) succeeded 2678d363e9540203fd34f78fe995c9935f6fc4f8
* :broken_heart: [Build #149195](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149195/) failed de50f5530718e2e0a40df8a160d8f5e87f66354d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149195/","number":"149195","commit":"de50f5530718e2e0a40df8a160d8f5e87f66354d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/","number":"149253","commit":"2678d363e9540203fd34f78fe995c9935f6fc4f8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/","number":"149271","commit":"5fa51487557485d4802d341cebc3a0519a7be38c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/","number":"149300","commit":"68d6105bc7ed0cea9b3f1e51e412a66385cfac4e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/","number":"149404","commit":"cbc24655478fc6fe624ba1eda0d01c39c01e270a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/","number":"149464","commit":"cf11c333fe4f1e232207c366e8b4163b8f5c8c76"}]}
PIPELINE-->

sqren commented:
@elasticmachine merge upstream

kibanamachine commented:
merge conflict between base and head

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/)
* Commit: e90d409c0d277d1de31760fc9c2a92745616d28c
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Build Kibana](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/execution/node/371/log)

### History
* :broken_heart: [Build #149464](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/) failed cf11c333fe4f1e232207c366e8b4163b8f5c8c76
* :broken_heart: [Build #149404](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/) failed cbc24655478fc6fe624ba1eda0d01c39c01e270a
* :broken_heart: [Build #149300](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/) failed 68d6105bc7ed0cea9b3f1e51e412a66385cfac4e
* :green_heart: [Build #149271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/) succeeded 5fa51487557485d4802d341cebc3a0519a7be38c
* :green_heart: [Build #149253](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/) succeeded 2678d363e9540203fd34f78fe995c9935f6fc4f8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149253/","number":"149253","commit":"2678d363e9540203fd34f78fe995c9935f6fc4f8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/","number":"149271","commit":"5fa51487557485d4802d341cebc3a0519a7be38c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/","number":"149300","commit":"68d6105bc7ed0cea9b3f1e51e412a66385cfac4e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/","number":"149404","commit":"cbc24655478fc6fe624ba1eda0d01c39c01e270a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/","number":"149464","commit":"cf11c333fe4f1e232207c366e8b4163b8f5c8c76"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/","number":"151380","commit":"e90d409c0d277d1de31760fc9c2a92745616d28c"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/)
* Commit: aa001d1366523977eddac899f84da09ea17bb6b8
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Build Kibana](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/execution/node/375/log)

### History
* :broken_heart: [Build #151380](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/) failed e90d409c0d277d1de31760fc9c2a92745616d28c
* :broken_heart: [Build #149464](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/) failed cf11c333fe4f1e232207c366e8b4163b8f5c8c76
* :broken_heart: [Build #149404](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/) failed cbc24655478fc6fe624ba1eda0d01c39c01e270a
* :broken_heart: [Build #149300](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/) failed 68d6105bc7ed0cea9b3f1e51e412a66385cfac4e
* :green_heart: [Build #149271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/) succeeded 5fa51487557485d4802d341cebc3a0519a7be38c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149271/","number":"149271","commit":"5fa51487557485d4802d341cebc3a0519a7be38c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/","number":"149300","commit":"68d6105bc7ed0cea9b3f1e51e412a66385cfac4e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/","number":"149404","commit":"cbc24655478fc6fe624ba1eda0d01c39c01e270a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/","number":"149464","commit":"cf11c333fe4f1e232207c366e8b4163b8f5c8c76"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/","number":"151380","commit":"e90d409c0d277d1de31760fc9c2a92745616d28c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/","number":"151408","commit":"aa001d1366523977eddac899f84da09ea17bb6b8"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/)
* Commit: 1c20a19f19572d31d717d8aa68e9a8a41d9d4b72
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109812/1c20a19f19572d31d717d8aa68e9a8a41d9d4b72)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/9b41b3feaee1a2dcdb8e9adb9c0aec4492006616) | [after](https://github.com/elastic/kibana/commit/1c20a19f19572d31d717d8aa68e9a8a41d9d4b72) | diff |
| --- | --- | --- | --- |
| `apm` | 41 | 42 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9b41b3feaee1a2dcdb8e9adb9c0aec4492006616) | [after](https://github.com/elastic/kibana/commit/1c20a19f19572d31d717d8aa68e9a8a41d9d4b72) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +375.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/9b41b3feaee1a2dcdb8e9adb9c0aec4492006616) | [after](https://github.com/elastic/kibana/commit/1c20a19f19572d31d717d8aa68e9a8a41d9d4b72) | diff |
| --- | --- | --- | --- |
| `apm` | 41 | 42 | +1 |


</details>

### History
* :broken_heart: [Build #151408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/) failed aa001d1366523977eddac899f84da09ea17bb6b8
* :broken_heart: [Build #151380](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/) failed e90d409c0d277d1de31760fc9c2a92745616d28c
* :broken_heart: [Build #149464](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/) failed cf11c333fe4f1e232207c366e8b4163b8f5c8c76
* :broken_heart: [Build #149404](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/) failed cbc24655478fc6fe624ba1eda0d01c39c01e270a
* :broken_heart: [Build #149300](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/) failed 68d6105bc7ed0cea9b3f1e51e412a66385cfac4e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149300/","number":"149300","commit":"68d6105bc7ed0cea9b3f1e51e412a66385cfac4e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/","number":"149404","commit":"cbc24655478fc6fe624ba1eda0d01c39c01e270a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/","number":"149464","commit":"cf11c333fe4f1e232207c366e8b4163b8f5c8c76"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/","number":"151380","commit":"e90d409c0d277d1de31760fc9c2a92745616d28c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/","number":"151408","commit":"aa001d1366523977eddac899f84da09ea17bb6b8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/","number":"151447","commit":"1c20a19f19572d31d717d8aa68e9a8a41d9d4b72"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/)
* Commit: 5034a4561df89df4996c30b43b8f2947f52abd2c
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/execution/node/430/log)

### History
* :green_heart: [Build #151447](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/) succeeded 1c20a19f19572d31d717d8aa68e9a8a41d9d4b72
* :broken_heart: [Build #151408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/) failed aa001d1366523977eddac899f84da09ea17bb6b8
* :broken_heart: [Build #151380](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/) failed e90d409c0d277d1de31760fc9c2a92745616d28c
* :broken_heart: [Build #149464](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/) failed cf11c333fe4f1e232207c366e8b4163b8f5c8c76
* :broken_heart: [Build #149404](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/) failed cbc24655478fc6fe624ba1eda0d01c39c01e270a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149404/","number":"149404","commit":"cbc24655478fc6fe624ba1eda0d01c39c01e270a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/149464/","number":"149464","commit":"cf11c333fe4f1e232207c366e8b4163b8f5c8c76"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/","number":"151380","commit":"e90d409c0d277d1de31760fc9c2a92745616d28c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/","number":"151408","commit":"aa001d1366523977eddac899f84da09ea17bb6b8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/","number":"151447","commit":"1c20a19f19572d31d717d8aa68e9a8a41d9d4b72"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/","number":"151490","commit":"5034a4561df89df4996c30b43b8f2947f52abd2c"}]}
PIPELINE-->

sqren commented:
@formgeist @dgieselaar Feedback addressed.

formgeist commented on x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/Waterfall/waterfall_item.tsx:
```diff
@@ -216,27 +212,52 @@ export function WaterfallItem({
+    return (
+      // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+      <div onClick={(e: React.MouseEvent) => e.stopPropagation()}>
+        <EuiBadge href={href} color={theme.eui.euiColorDanger}>
```
```suggestion
        <EuiBadge href={href} color={theme.eui.euiColorDanger} iconType="arrowRight">
```

I suggest that we add another visual differentiator to this badge because it's a link and not just a display.

<img width="521" alt="CleanShot 2021-09-08 at 10 30 50@2x" src="https://user-images.githubusercontent.com/4104278/132475033-07aff2bf-5278-4848-83a3-3997699f8434.png">

formgeist reviewed (commented):
@sqren Looks great - I have a small suggestion to enhance the "View related error" badge button.

Also, I found that the failure and other badges don't vertically align.

<img width="317" alt="CleanShot 2021-09-08 at 10 30 42@2x" src="https://user-images.githubusercontent.com/4104278/132475209-99ace786-487f-4b88-820b-93ea53575a0d.png">

sorenlouv committed:
Add `arrowRight`

Co-authored-by: Casper Hübertz <casper@formgeist.com>

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151738/)
* Commit: 689ba2fd6ae2fabccad7e51cb00454a3e285f5d6
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Quick Commit Checks](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151738/execution/node/419/log)

### History
* :green_heart: [Build #151666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151666/) succeeded ff1e48af2dfb5a33db9421929b59680ca6f1c88a
* :broken_heart: [Build #151490](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/) failed 5034a4561df89df4996c30b43b8f2947f52abd2c
* :green_heart: [Build #151447](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/) succeeded 1c20a19f19572d31d717d8aa68e9a8a41d9d4b72
* :broken_heart: [Build #151408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/) failed aa001d1366523977eddac899f84da09ea17bb6b8
* :broken_heart: [Build #151380](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/) failed e90d409c0d277d1de31760fc9c2a92745616d28c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151380/","number":"151380","commit":"e90d409c0d277d1de31760fc9c2a92745616d28c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/","number":"151408","commit":"aa001d1366523977eddac899f84da09ea17bb6b8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/","number":"151447","commit":"1c20a19f19572d31d717d8aa68e9a8a41d9d4b72"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/","number":"151490","commit":"5034a4561df89df4996c30b43b8f2947f52abd2c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151666/","number":"151666","commit":"ff1e48af2dfb5a33db9421929b59680ca6f1c88a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151738/","number":"151738","commit":"689ba2fd6ae2fabccad7e51cb00454a3e285f5d6"}]}
PIPELINE-->

sorenlouv committed:
Apply prettier

sorenlouv committed:
Improve spacing and vertical alignment

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151811/)
* Commit: 507241f424f29f8fb6313215e257cea4b54b1218
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109812/507241f424f29f8fb6313215e257cea4b54b1218)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/f1fbe8e06de510e59623e03fa46552c530072ef7) | [after](https://github.com/elastic/kibana/commit/507241f424f29f8fb6313215e257cea4b54b1218) | diff |
| --- | --- | --- | --- |
| `apm` | 41 | 42 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f1fbe8e06de510e59623e03fa46552c530072ef7) | [after](https://github.com/elastic/kibana/commit/507241f424f29f8fb6313215e257cea4b54b1218) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +311.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/f1fbe8e06de510e59623e03fa46552c530072ef7) | [after](https://github.com/elastic/kibana/commit/507241f424f29f8fb6313215e257cea4b54b1218) | diff |
| --- | --- | --- | --- |
| `apm` | 41 | 42 | +1 |


</details>

### History
* :broken_heart: [Build #151738](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151738/) failed 689ba2fd6ae2fabccad7e51cb00454a3e285f5d6
* :green_heart: [Build #151666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151666/) succeeded ff1e48af2dfb5a33db9421929b59680ca6f1c88a
* :broken_heart: [Build #151490](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/) failed 5034a4561df89df4996c30b43b8f2947f52abd2c
* :green_heart: [Build #151447](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/) succeeded 1c20a19f19572d31d717d8aa68e9a8a41d9d4b72
* :broken_heart: [Build #151408](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/) failed aa001d1366523977eddac899f84da09ea17bb6b8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151408/","number":"151408","commit":"aa001d1366523977eddac899f84da09ea17bb6b8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/","number":"151447","commit":"1c20a19f19572d31d717d8aa68e9a8a41d9d4b72"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/","number":"151490","commit":"5034a4561df89df4996c30b43b8f2947f52abd2c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151666/","number":"151666","commit":"ff1e48af2dfb5a33db9421929b59680ca6f1c88a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151738/","number":"151738","commit":"689ba2fd6ae2fabccad7e51cb00454a3e285f5d6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151811/","number":"151811","commit":"507241f424f29f8fb6313215e257cea4b54b1218"}]}
PIPELINE-->

sorenlouv committed:
[APM] Show badge for failed spans

Fix i18n

[APM] Indicate failure on spans

fix i18n

add api test

fix setup script and type fixes

Add `event.outcome` to flyout metadata

Fix tsc issues

Re-order badges

Fix tsc

sqren commented:
> Also, I found that the failure and other badges don't vertically align.


Good catch. Fixed.

formgeist reviewed (approved):
LGTM 👍

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151962/)
* Commit: 95d3cda74d4ae52d9d6d0af2af904581a86640f8
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109812/95d3cda74d4ae52d9d6d0af2af904581a86640f8)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/178b56980f5744339cd7344d57ef529836a32172) | [after](https://github.com/elastic/kibana/commit/95d3cda74d4ae52d9d6d0af2af904581a86640f8) | diff |
| --- | --- | --- | --- |
| `apm` | 41 | 42 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/178b56980f5744339cd7344d57ef529836a32172) | [after](https://github.com/elastic/kibana/commit/95d3cda74d4ae52d9d6d0af2af904581a86640f8) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +311.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/178b56980f5744339cd7344d57ef529836a32172) | [after](https://github.com/elastic/kibana/commit/95d3cda74d4ae52d9d6d0af2af904581a86640f8) | diff |
| --- | --- | --- | --- |
| `apm` | 41 | 42 | +1 |


</details>

### History
* :green_heart: [Build #151811](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151811/) succeeded 507241f424f29f8fb6313215e257cea4b54b1218
* :broken_heart: [Build #151738](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151738/) failed 689ba2fd6ae2fabccad7e51cb00454a3e285f5d6
* :green_heart: [Build #151666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151666/) succeeded ff1e48af2dfb5a33db9421929b59680ca6f1c88a
* :broken_heart: [Build #151490](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/) failed 5034a4561df89df4996c30b43b8f2947f52abd2c
* :green_heart: [Build #151447](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/) succeeded 1c20a19f19572d31d717d8aa68e9a8a41d9d4b72

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151447/","number":"151447","commit":"1c20a19f19572d31d717d8aa68e9a8a41d9d4b72"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151490/","number":"151490","commit":"5034a4561df89df4996c30b43b8f2947f52abd2c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151666/","number":"151666","commit":"ff1e48af2dfb5a33db9421929b59680ca6f1c88a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151738/","number":"151738","commit":"689ba2fd6ae2fabccad7e51cb00454a3e285f5d6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151811/","number":"151811","commit":"507241f424f29f8fb6313215e257cea4b54b1218"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151962/","number":"151962","commit":"95d3cda74d4ae52d9d6d0af2af904581a86640f8"}]}
PIPELINE-->

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/111652)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/111652">](https://github.com/elastic/kibana/pull/111652) |

This backport PR will be merged automatically after passing CI.
